src/arns.lua

```diff
@@ -664,7 +664,7 @@ function arns.getTokenCost(intendedAction)
 		-- stub the process id as it is not required for this intent
 		local processId = "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
 		arns.assertValidBuyRecord(name, years, purchaseType, processId, false)
-		tokenCost = arns.calculateRegistrationFee(purchaseType, baseFee, years, demand.getDemandFactor())
+		tokenCost = arns.calculateRegistrationFee(purchaseType or "lease", baseFee, years, demand.getDemandFactor())
```

Comment:
Might make sense to hoist the `or 'lease'` up to the assignment to purchaseType to avoid potential mismatches if used elsewhere in the function.

Reply:
ah, missed that - [c229f56](https://github.com/ar-io/ar-io-network-process/pull/328/commits/c229f56199a82931cd0951f2efbc643a6add8525)